pkg/proxy/backend/authenticator.go

```diff
@@ -156,23 +156,29 @@ func (auth *Authenticator) handshakeFirstTime(logger *zap.Logger, cctx ConnConte
 	// In case of testing, backendIO is passed manually that we don't want to bother with the routing logic.
 	backendIO, err := getBackendIO(cctx, auth, clientResp, 5*time.Second)
 	if err != nil {
-		return err
+		return WrapUserError(err, connectErrMsg)
```

Comment:
This basically wraps all errors in `getBackendIO` to `connectErrMsg`.

While the underlying errors may contain messages from `handshakeHandler.GetRouter()`, `errors.As` will stop at the most outside error, i.e. the error wrapped this line. We should not wrap user errors multiple times:  invoke `errors.As` on`Tiproxy wrapped -> backoff wrapped -> handler wrapped -> real msg` will stop at `Tiproxy wrapped`.

Reply:
I checked it in `WrapUserError`:
```go
if ue, ok := err.(*UserError); ok {
	return ue
}
```

Reply:
What if `WrapUserError(errors.Errorf("%w", WrapUserError())))`? It can be a long chain since we can't decide how gateway use this function. Either we just don't export the function, or we just let them call, IMO.

Reply:
If this is a problem, I can replace `err.(*UserError)` with `errors.As`.